Core/include/Acts/EventData/MultiTrajectory.ipp

```diff
@@ -77,20 +77,15 @@ inline auto TrackStateProxy<D, M, ReadOnly>::projector() const -> Projector {
 
 template <typename D, size_t M, bool ReadOnly>
 inline auto TrackStateProxy<D, M, ReadOnly>::uncalibratedSourceLink() const
-    -> const SourceLink& {
-  assert(has<hashString("uncalibratedSourceLink")>());
-  return component<std::optional<SourceLink>,
-                   hashString("uncalibratedSourceLink")>()
-      .value();
+    -> SourceLink {
+  return getUncalibratedSourceLink();
 }
 
 template <typename D, size_t M, bool ReadOnly>
-inline auto TrackStateProxy<D, M, ReadOnly>::calibratedSourceLink() const
-    -> const SourceLink& {
-  assert(has<hashString("calibratedSourceLink")>());
-  return component<std::optional<SourceLink>,
-                   hashString("calibratedSourceLink")>()
-      .value();
+inline auto TrackStateProxy<D, M, ReadOnly>::getUncalibratedSourceLink() const
+    -> SourceLink {
+  assert(has<hashString("uncalibratedSourceLink")>());
+  return m_traj->getUncalibratedSourceLink(m_istate);
```

Comment:
Misisng `_impl`?

Reply:
This is still in the MTJ, there's a function in the `.hpp` that in turn calls the `_impl()` method.

Reply:
A, yes, this is in proxy. This header is quite long ...